cybershell/vulnerability_kb.py

```
@@ -0,0 +1,740 @@
+            "exploitation_sequences": [],
+            "detection_patterns": {},
+            "bypass_strategies": [],
+            "product_specific_data": {}  # NEW
+        }
+        
+        # Organize patterns by category
+        for category in VulnCategory:
+            payloads = self.get_payloads_by_category(category)
+            training_data["vulnerability_patterns"][category.value] = {
+                "payloads": [p.payload for p in payloads],
+                "contexts": list(set(p.context for p in payloads if p.context)),
```

Ruff (0.12.2)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)